src/odoc/bin/main.ml

```diff
@@ -137,18 +128,28 @@ end = struct
     in
     Arg.(value & opt_all string default & info ~docv:"MODULE" ~doc ["open"])
 
+  let children =
+    let doc = "Specify the odoc file as a child. Can be used multiple times. Only applies to mld files" in
+    let default = [] in
+    Arg.(value & opt_all string default & info ~docv:"CHILD" ~doc ["child"])
+
   let cmd =
-    let pkg =
-      let doc = "Package the input is part of" in
-      Arg.(required & opt (some string) None &
+    let package_opt =
+      let doc = "Package the input is part of (deprecated - use '--parent')" in
+      Arg.(value & opt (some string) None &
            info ~docs ~docv:"PKG" ~doc ["package"; "pkg"])
     in
+    let parent_opt =
+      let doc = "Parent page or subpage" in
+      Arg.(value & opt (some string) None &
+           info ~docs ~docv:"PARENT" ~doc ["parent"])
+    in
```

Comment:
Both options don't work at the same time, `--package` would be ignored if `--parent` is passed. Maybe we should resolve this here so `Compile` don't have to decide what to do with these at all:
```ocaml
let parent_opt =
  let package_opt = ...
  let parent_opt = ...
  let one_of pkg parent =
    match pkg, parent with
    | Some _, Some _ -> (* An error here *)
    | Some pkg, None -> Some (`Package pkg)
    | None, Some parent -> Some (`Parent parent)
    | None, None -> None
  in
  Term.(const one_of $ package_opt $ parent_opt)
in
```

Reply:
Yeah that's a good idea!